fabric-resource-loader-v0/src/test/java/net/fabricmc/fabric/test/resource/loader/unit/ModResourcePackUtilTests.java

```diff
@@ -0,0 +1,248 @@
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.test.resource.loader.unit;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeMap;
+import java.util.function.Predicate;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
+import org.jetbrains.annotations.Nullable;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import net.minecraft.Bootstrap;
+import net.minecraft.SharedConstants;
+import net.minecraft.resource.ResourcePackProfile;
+
+import net.fabricmc.fabric.impl.resource.loader.FabricResourcePackProfile;
+import net.fabricmc.fabric.impl.resource.loader.ModResourcePackCreator;
+import net.fabricmc.fabric.impl.resource.loader.ModResourcePackUtil;
+
+public class ModResourcePackUtilTests {
+	private static final Gson GSON = new Gson();
+
+	@BeforeAll
+	static void beforeAll() {
+		SharedConstants.createGameVersion();
+		Bootstrap.initialize();
+	}
+
+	@Test
+	void testRefreshAutoEnabledPacks() {
+		// Vanilla uses tree map, and we test the behavior
+		Map<String, ResourcePackProfile> profiles = new TreeMap<>();
+		Map<String, ResourcePackProfile> modAProfiles = new TreeMap<>();
+		Map<String, ResourcePackProfile> modBProfiles = new TreeMap<>();
+		Map<String, ResourcePackProfile> allProfiles = new TreeMap<>();
+		ResourcePackProfile vanilla = mockProfile(profiles, "vanilla", null);
+		ResourcePackProfile fabric = mockProfile(profiles, ModResourcePackCreator.FABRIC, null);
+		ResourcePackProfile modA = mockProfile(modAProfiles, "mod_a", ModResourcePackCreator.BASE_PARENT);
+		ResourcePackProfile modAProg = mockProfile(modAProfiles, "mod_a_programmer_art", ModResourcePackCreator.PROGRAMMER_ART_PARENT);
+		ResourcePackProfile modAHigh = mockProfile(modAProfiles, "mod_a_high_contrast", ModResourcePackCreator.HIGH_CONTRAST_PARENT);
+		ResourcePackProfile modB = mockProfile(modBProfiles, "mod_b", ModResourcePackCreator.BASE_PARENT);
+		ResourcePackProfile modBProg = mockProfile(modBProfiles, "mod_b_programmer_art", ModResourcePackCreator.PROGRAMMER_ART_PARENT);
+		ResourcePackProfile modBHigh = mockProfile(modBProfiles, "mod_b_high_contrast", ModResourcePackCreator.HIGH_CONTRAST_PARENT);
+		ResourcePackProfile programmerArt = mockProfile(profiles, "programmer_art", null);
+		ResourcePackProfile highContrast = mockProfile(profiles, "high_contrast", null);
+		ResourcePackProfile userPackA = mockProfile(profiles, "user_pack_a", null);
+		ResourcePackProfile userPackB = mockProfile(profiles, "user_pack_b", null);
+		modAProfiles.putAll(profiles);
+		modBProfiles.putAll(profiles);
+		allProfiles.putAll(modAProfiles);
+		allProfiles.putAll(modBProfiles);
+
+		testRefreshAutoEnabledPacks(
```

Comment:
Nit: You would usually have a test method for each of these, not the end of the world but would make it more clear if just one of these tests broke.

Happy to leave this be.